CONTRIBUTING.md

```diff
@@ -64,7 +64,7 @@ requests (PRs).
 To create a new PR, fork the project in GitHub and clone the upstream repo:
 
 ```sh
-git clone https://github.com/open-telemetry/opentelemetry-cpp.git
+git clone --recursive https://github.com/open-telemetry/opentelemetry-cpp.git
```

Comment:
nit: Just want to be sure that the changes done in COTRIBUTING.md as part of #416 are still relevant after this ?

Reply:
https://github.com/open-telemetry/opentelemetry-cpp/pull/416 says "if you haven't, make sure loading submodules` so the commands are not required if submodules are available. I am fine with either keeping it or removing the submodule commands in #416.